### PR TITLE
Merge in Registry Features

### PR DESCRIPTION
**Pull request by darinacosta**
This branch is a merge of Terranodo's Registry, Hypermap, and Hypersearch additions, and also includes the following bug fixes:

- The "zoom to extent" functionality works for registry layers.
- The cross domain error caused by including a GeoNode CSRF token in the request header to the default hypermap catalogue has been addressed.
- The application does not error out when embedded in a GeoNode session. 
- GeoNode layers can be added to the map.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
-before_install:
-  - mkdir travis-phantomjs
-  - wget https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-2.1.1-linux-x86_64.tar.bz2 -O $PWD/travis-phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2
-  - tar -xvf $PWD/travis-phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2 -C $PWD/travis-phantomjs
-  - export PATH=$PWD/travis-phantomjs/phantomjs-2.1.1-linux-x86_64/bin:$PATH
-
+- '0.12'
+- '0.10'
+addons:
+  code_climate:
```
Comment (BerryDaniel) [inline .travis.yml]: lets remove codeclimate, since it is tied to a specific account

File: .travis.yml [hunk 2]
```diff
@@ -1,20 +1,25 @@
...
-  - npm install --quiet -g grunt-cli karma bower
-  - bower install
-  - grunt
-  - grunt karmaconfig
-
-script: grunt karma
-
+- export DISPLAY=:99.0
+- sh -e /etc/init.d/xvfb start
+- npm install --quiet -g grunt-cli karma bower codeclimate-test-reporter
```
Comment (BerryDaniel) [inline .travis.yml]: no need to install codeclimate-test-reporter

File: .travis.yml [hunk 3]
```diff
@@ -1,20 +1,25 @@
...
+- bower install
+- grunt
+- grunt karmaconfig
+script: grunt test
+env:
+  matrix:
+  - DEPLOY_BRANCH=hypermap
```
Comment (BerryDaniel) [inline .travis.yml]: should not be tied to hypermap branch

File: .travis.yml [hunk 4]
```diff
@@ -1,20 +1,25 @@
...
+  global:
+  - secure: Abv5HJD1co7uWtYTxme2iNeoZNDqB7KuNSTBg7FCZtGObnzR1Y7kJsrjjaVG1YszL7uuX0tL423sc12wHdDy0+vMOgdPVT3dHF6lfKxFYd4e08NqStXirrCBH6SpY6s+RHd7KIUzqUyHq0nUrVa2pp/AnjkHpVhYNfjnlhsaCni1Rot7QkY2SfzjChQhemI5JWkHlr0hNEDhHHzPszed+Teh02frKKkR4FJSMaeStdQkOSAytGdOds1HwbXaVgjQYpZ2CPPb88FloXVCBYRyh28Vm2AaO/b/PudmWxxE4mWxJE0KGzf6PYZjl9sKUuSlDcdzxIrXo3+vuydni/HaO+GWS5ZwYPwn6Aoaqdy6OkKVMnloXbEaoLqDhXjH+Y3GKQ0IRzc16ygalZ/Hsa83GdI0136wPy6bpyif52yCRkr1FT8akvoY+Hmj/WFLI5XjgU9LEqldNajzjtrRkSEYXxnjQreQ4GQHNdQrz4ZXw4+O6W+4ARhLJCEs4O1KQoeJmLpz+blU/5VOpj17O767OATehgelrvFO4sOVXEC/IW5Jw7+l0zq3NylGfrg7Z1N0QXuw1uqwJYt3YodIWwCUV2FxUbDN3LckcOo4iP6zlwelBiyR1wE25TUe+58XC5EzyH6kvykK6gQky/oZ2i74HKv6pmOY1f0AzsLeATJfkcc=
+after_success:
+- grunt deploy
+after_script:
+- codeclimate-test-reporter < reports/coverage/lcov.info
```
Comment (BerryDaniel) [inline .travis.yml]: remove

File: Gruntfile.js [hunk 5]
```diff
@@ -10,16 +10,16 @@ module.exports = function ( grunt ) {
...
   grunt.loadNpmTasks('grunt-bump');
-  grunt.loadNpmTasks('grunt-coffeelint');
   grunt.loadNpmTasks('grunt-recess');
   grunt.loadNpmTasks('grunt-karma');
   grunt.loadNpmTasks('grunt-newer');
   grunt.loadNpmTasks('grunt-ngmin');
   grunt.loadNpmTasks('grunt-html2js');
   grunt.loadNpmTasks('grunt-gjslint');
+  grunt.loadNpmTasks('grunt-contrib-connect');
+  grunt.loadNpmTasks('grunt-gh-pages');
```
Comment (BerryDaniel) [inline Gruntfile.js]: no need to deploy to gh-pages

File: Gruntfile.js [hunk 6]
```diff
@@ -37,6 +37,35 @@ module.exports = function ( grunt ) {
...
+    connect: {
+      server: {
+        options: {
+          port: 9001,
+          base: 'build'
+        }
+      }
+    },
+
+    'gh-pages': {
```
Comment (BerryDaniel) [inline Gruntfile.js]: remove, especially since it is tied to a repo and user

File: Gruntfile.js [hunk 7]
```diff
@@ -777,5 +748,41 @@ module.exports = function ( grunt ) {
...
+
+  grunt.registerTask('check-deploy', function() {
+    // need this
+    this.requires(['build']);
+
+    // only deploy under these conditions
+    if (process.env.TRAVIS === 'true' && process.env.TRAVIS_SECURE_ENV_VARS === 'true' && process.env.TRAVIS_PULL_REQUEST === 'false' && process.env.TRAVIS_BRANCH == process.env.DEPLOY_BRANCH) {
+      grunt.log.writeln('executing deployment');
+      // queue deploy
+      grunt.task.run('gh-pages:deploy');
```
Comment (BerryDaniel) [inline Gruntfile.js]: remove

File: README.md [hunk 8]
```diff
@@ -1,16 +1,13 @@
-[![Build Status](https://travis-ci.org/ROGUE-JCTD/MapLoom.svg?branch=master)](https://travis-ci.org/ROGUE-JCTD/MapLoom)
-
-MapLoom
-=========
-
+MapLoom [![Build Status](https://travis-ci.org/terranodo/MapLoom.svg?branch=hypermap)](https://travis-ci.org/terranodo/MapLoom) [![Code Climate](https://codeclimate.com/github/terranodo/MapLoom/badges/gpa.svg)](https://codeclimate.com/github/terranodo/MapLoom) [![Test Coverage](https://codeclimate.com/github/terranodo/MapLoom/badges/coverage.svg)](https://codeclimate.com/github/terranodo/MapLoom/coverage)
```
Comment (BerryDaniel) [inline README.md]: revert back to ROGUE-JCTD or use one for Boundless

File: README.md [hunk 9]
```diff
@@ -45,6 +42,17 @@ The recommended way to develop maploom is to:
    ```
    geoshape-config maploom_dev
    ```
-- you can now go to http://geoshape_vm_server_ip/maps/new or http://geoshape_vm_server_ip/layers to create a map from a layer. Changes on your host machine will cause `grunt watch` to rebuild MapLoom, and the symbolic links created by `geoshape-config maploom_dev` will make the newly buit maploom immediately available to Geonode in your VM. 
+- you can now go to http://geoshape_vm_server_ip/maps/new or http://geoshape_vm_server_ip/layers to create a map from a layer. Changes on your host machine will cause `grunt watch` to rebuild MapLoom, and the symbolic links created by `geoshape-config maploom_dev` will make the newly buit maploom immediately available to Geonode in your VM.
+
+Note that linter used by `grunt watch` is very paticular about the js programing style and guildlines. Be sure to fix all compile issues.
+
+## Deployment
```
Comment (BerryDaniel) [inline README.md]: remove gh-pages

File: src/common/addlayers/ServerService.js [hunk 10]
```diff
@@ -418,27 +458,373 @@ var SERVER_SERVICE_USE_PROXY = true;
...
+    var createHyperSearchLayerObject = function(layerInfo, serverUrl) {
+      return {
+        add: true,
+        Abstract: layerInfo.Abstract,
+        Name: layerInfo.LayerName,
+        Title: layerInfo.LayerTitle,
+        LayerDate: layerInfo.LayerDate,
+        LayerCategory: layerInfo.LayerCategory,
+        CRS: ['EPSG:4326'],
+        detail_url: 'http://52.38.116.143/layer/' + layerInfo.LayerId,
```
Comment (BerryDaniel) [inline src/common/addlayers/ServerService.js]: Why is this hardcoded? We need a way to pass the registry url

File: src/common/addlayers/ServerService.spec.js [hunk 11]
```diff
@@ -1,116 +1,380 @@
...
+        expect(serverService.reformatLayerConfigs({objects: [] }, '').length).toEqual(0);
+      });
+    });
+    describe('result has one layer', function() {
+      var layers = {};
+      beforeEach(function() {
+        layers.objects = [
+          {
+            title: 'Ocean Beach',
+            detail_url: '/layers/OceanBeach',
```
Comment (BerryDaniel) [inline src/common/addlayers/ServerService.spec.js]: What is this? Is it expecting this layer?

File: src/common/addlayers/ServerService.spec.js [hunk 12]
```diff
@@ -1,116 +1,380 @@
...
+            }
+          }
+        ];
+      });
+      it('returns one formatted layer', function() {
+        expect(serverService.reformatLayerHyperConfigs(layers, '').length).toEqual(1);
+      });
+      it('has a thumbnail image', function() {
+        expect(serverService.reformatLayerHyperConfigs(layers, '')[0]).toEqual(jasmine.objectContaining({
+          thumbnail_url: 'http://52.38.116.143/test.jpg'
```
Comment (BerryDaniel) [inline src/common/addlayers/ServerService.spec.js]: hardcoded ip

File: src/common/addlayers/ServerService.spec.js [hunk 13]
```diff
@@ -1,116 +1,380 @@
...
+        configService.username = 'Dijkstra';
+      });
+      it('returns the url with q', function() {
+        var filterOptions = {
+          owner: true,
+          text: null,
+          from: null,
+          size: null
+        };
+        expect(serverService.applyESFilter('mapstory', filterOptions)).toEqual('mapstory&owner__username__in=Dijkstra');
```
Comment (BerryDaniel) [inline src/common/addlayers/ServerService.spec.js]: Dijkstra entries in code base should be removed.

File: src/common/configuration/ConfigService.js [hunk 14]
```diff
@@ -52,32 +52,36 @@
...
           projection: 'EPSG:900913',
           center: [-9707182.048613328, 1585691.7893914054],
-          zoom: 14,
+          zoom: 1,
           layers: [
             {
               'opacity': 1,
               'selected': true,
               'group': 'background',
-              'name': 'mapnik',
```
Comment (BerryDaniel) [inline src/common/configuration/ConfigService.js]: revert back to mapnik

File: src/common/configuration/ConfigService.js [hunk 15]
```diff
@@ -52,32 +52,36 @@
...
-              'type': 'OpenLayers.Layer.OSM'
+              'type': 'OpenLayers.Layer.OSM',
+              'sourceParams': { 'layer': 'world-light' }
             }
           ]
         },
         sources: [
           {
-            'url': ('http://' + $location.host() + '/geoserver/wms'),
+            'url': ('http://geoshape.geointservices.io/geoserver/web/'),
```
Comment (BerryDaniel) [inline src/common/configuration/ConfigService.js]: do not use hardcoded geointservices.io entry

File: src/common/configuration/ConfigService.js [hunk 16]
```diff
@@ -87,13 +91,25 @@
         authStatus: 401,
         id: 0,
         proxy: '/proxy/?url=',
+        registryEnabled: true,
         nominatimUrl: 'http://nominatim.openstreetmap.org',
         fileserviceUrlTemplate: '/api/fileservice/view/{}',
-        fileserviceUploadUrl: '/api/fileservice/'
+        fileserviceUploadUrl: '/api/fileservice/',
+        catalogList: [
+          {name: 'hypersearch catalog 1', url: 'http://geoshape.geointservices.io/search/hypermap/'},
```
Comment (BerryDaniel) [inline src/common/configuration/ConfigService.js]: catalog list should be an array passed from django

File: src/common/map/MapDirective.js [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+          link: function(scope, element) {
+            var map;
+            var firstExtent = null;
+
+            var createMap = function() {
+              map = new ol.Map({
+                layers: [
+                  new ol.layer.Tile({
+                    source: new ol.source.TileJSON({
+                      url: 'http://api.tiles.mapbox.com/v4/mapbox.world-light.json?access_token=pk.eyJ1IjoiYmVja2VyciIsImEiOiJjaWtzcHVyeTYwMDA3dWdsenB5aHUxMzl1In0.1FVjOTdhoXGXtnfApX8wVQ',
```
Comment (BerryDaniel) [inline src/common/map/MapDirective.js]: whose access token is this, we should revert to osm

File: src/common/addlayers/ServerService.spec.js (hunk 11 above)
Comment (darinacosta) [inline src/common/addlayers/ServerService.spec.js]: @BerryDaniel It's being used for testing.

File: src/common/map/MapDirective.js (hunk 17 above)
Comment (darinacosta) [inline src/common/map/MapDirective.js]: @BerryDaniel resolved

File: .travis.yml (hunk 1 above)
Comment (darinacosta) [inline .travis.yml]: @BerryDaniel resolved

File: .travis.yml (hunk 2 above)
Comment (darinacosta) [inline .travis.yml]: @BerryDaniel resolved

File: .travis.yml (hunk 3 above)
Comment (darinacosta) [inline .travis.yml]: @BerryDaniel resolved

File: .travis.yml (hunk 4 above)
Comment (darinacosta) [inline .travis.yml]: @BerryDaniel resolved

File: Gruntfile.js (hunk 5 above)
Comment (darinacosta) [inline Gruntfile.js]: @BerryDaniel resolved

File: Gruntfile.js (hunk 6 above)
Comment (darinacosta) [inline Gruntfile.js]: @BerryDaniel resolved

File: Gruntfile.js (hunk 7 above)
Comment (darinacosta) [inline Gruntfile.js]: resolved

File: README.md (hunk 8 above)
Comment (darinacosta) [inline README.md]: resolved

File: README.md (hunk 9 above)
Comment (darinacosta) [inline README.md]: resolved

File: src/common/addlayers/ServerService.spec.js (hunk 13 above)
Comment (darinacosta) [inline src/common/addlayers/ServerService.spec.js]: A mock for a jasmine test.

File: src/common/configuration/ConfigService.js (hunk 16 above)
Comment (darinacosta) [inline src/common/configuration/ConfigService.js]: This will be overwritten by Django configs.

**Pull request closed without merging**